staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight_test.go

```diff
@@ -210,7 +211,67 @@ func TestMaxInFlightMutating(t *testing.T) {
 	// Show that we recover from being blocked up.
 	// Too avoid flakyness we need to wait until at least one of the requests really finishes.
 	responses.Wait()
-	if err := expectHTTPPost(server.URL+"/foo/bar", http.StatusOK); err != nil {
+	if err := expectHTTPPost(server.URL+"/foo/bar", http.StatusOK, false); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestMaxInFlightDisconnect(t *testing.T) {
+	const AllowedMutatingInflightRequestsNo = 3
+
+	calls := &sync.WaitGroup{}
+	calls.Add(AllowedMutatingInflightRequestsNo)
+
+	responses := &sync.WaitGroup{}
+	responses.Add(AllowedMutatingInflightRequestsNo)
+
+	// Block is used to keep requests in flight for as long as we need to. All requests will
+	// be unblocked at the same time.
+	block := &sync.WaitGroup{}
+	block.Add(1)
+
+	waitForCalls := true
+	waitForCallsMutex := sync.Mutex{}
+
+	server := createMaxInflightServer(calls, block, &waitForCalls, &waitForCallsMutex, 1, AllowedMutatingInflightRequestsNo, 1.0)
```

Comment:
Can you add a comment that the 1.0 at the end means that every 429 request will be closed?